test/httpClient.js

```diff
@@ -0,0 +1,106 @@
+const undici = require('undici');
+const FormData = require('form-data');
+
+/**
+ * @typedef {'GET' | 'PUT' | 'DELETE' | 'PATCH' | 'POST'} THttpMethod
+ */
+
+/**
+ * @typedef {Object.<string, string | string[] | undefined>} THttpHeaders
+ */
+
+/**
+ * @typedef {Object} THttpRequest
+ * @property {string} path
+ * @property {*} [body] - TODO: restrict
+ * @property {boolean} [streaming]
+ * @property {string} [auth]
+ * @property {THttpHeaders} [headers]
+ */
+
+/**
+ * @typedef {Object} THttpResponse
+ * @property {number} status
+ * @property {THttpHeaders} headers
+ * @property {THttpHeaders} trailers
+ * @property {*} body - TODO: restrict
+ */
+
+/**
+ * @typedef {Object} THttpClient
+ * @property {(req: THttpRequest) => Promise<THttpResponse>} get
+ * @property {(req: THttpRequest) => Promise<THttpResponse>} put
+ * @property {(req: THttpRequest) => Promise<THttpResponse>} delete
+ * @property {(req: THttpRequest) => Promise<THttpResponse>} post
+ * @property {(req: THttpRequest) => Promise<THttpResponse>} patch
+ */
+
+/**
+ * @param {string} user
+ * @param {string} pw
+ * @returns {string}
+ */
+const makeBasicAuthHeader = (user, pw) => {
+  return `Basic ${Buffer.from(`${user}:${pw}`).toString('base64')}`;
+};
+
+/**
+ * @param {THttpMethod} method
+ * @param {THttpRequest} req
+ * @param {undici.Client} undiciClient
+ * @returns {Promise<THttpResponse>}
+ */
+const handleHttpRequest = async (method, req, undiciClient) => {
+  const reqHeaders = {
+    'content-type': 'application/json; charset=utf-8',
+    ...(req.auth
+      ? { authorization: makeBasicAuthHeader(`${req.auth}@email.be`, 'pwd') }
+      : {}),
```

Comment:
Is this ever used?

Reply:
Yes, this is a necessary header to indicate the content-type of the http request. Without this headers requests will fail.

Reply:
I meant the `auth` part, that's always empty.